db/filters/base.py

```diff
@@ -169,4 +158,47 @@ def getSAFilterSpecFromPredicate(pred: Predicate) -> dict:
     Not,
     And,
     Or,
-    ]
+]
+
+def takesParameterThatsAMathesarType(predicateSubClass: Type[Predicate]) -> bool:
+    return (
+        issubclass(predicateSubClass, Leaf)
+        and not issubclass(predicateSubClass, NoParameter)
+    )
+
+def not_empty(l): return len(l) > 0
+
+def assertPredicateCorrect(predicate):
```

Comment:
@dmos62 we use snake case for Python functions and variables too. We only use camel case (capitalized) for class names.

Reply:
Fixed it. Thanks for pointing that out.